reuse.go

```diff
@@ -46,6 +75,51 @@ func newReuse() (*reuse, error) {
 	}, nil
 }
 
+func (r *reuse) runGarbageCollector() {
+	ticker := time.NewTicker(garbageCollectInterval)
+	for {
+		<-ticker.C
+
+		now := time.Now()
+		var shouldExit bool
+		r.mutex.Lock()
+		for key, conn := range r.global {
+			if conn.ShouldGarbageCollect(now) {
+				delete(r.global, key)
+			}
+		}
+		for ukey, conns := range r.unicast {
+			for key, conn := range conns {
+				if conn.ShouldGarbageCollect(now) {
+					delete(conns, key)
+				}
+			}
+			if len(conns) == 0 {
+				delete(r.unicast, ukey)
+			}
+		}
+
```

Comment:
Don't we need to close sockets somewhere in here?

Reply:
Right, that's the whole point of this PR 😬